### PR TITLE
COG-898 Represent internal exceptions as a response with code 500

### PR DESCRIPTION
**Pull request by dmytro-palczewski-osi**
### JIRA Issue Link
- [COG-898: Represent internal exceptions as a response with code 500](https://osi-cwds.atlassian.net/browse/COG-898)

### Technical Description
Currently, we are receiving the response with 401 code in case of some internal exception happened.
Represent internal exceptions as a response with code 500.

### Tests
- [x] I have included unit tests 
- [x] I have included integration tests 
- [ ] I have included performance tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

### Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/idm/service/IdmServiceImpl.java [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import static gov.ca.cwds.config.api.idm.Roles.COUNTY_ADMIN;
 import static gov.ca.cwds.config.api.idm.Roles.CWS_WORKER;
 import static gov.ca.cwds.config.api.idm.Roles.OFFICE_ADMIN;
+import static gov.ca.cwds.idm.IdmResource.DATETIME_FORMATTER;
```
Comment (sorych) [inline src/main/java/gov/ca/cwds/idm/service/IdmServiceImpl.java]: may this be moved to constants/util class?

File: src/main/resources/messages/user_messages.properties [hunk 2]
```diff
@@ -35,4 +35,5 @@ CAP-034=Unable to write last login time for the user, username: {0}
 CAP-035=User's first name is not provided
 CAP-036=User's last name is not provided
 CAP-037=User's county name is not provided
-CAP-038=Unable to write last registration resubmit time for the user, username: {0}
\ No newline at end of file
+CAP-038=Unable to write last registration resubmit time for the user, username: {0}
+CAP-039=Something went wrong in Cognito
```
Comment (sorych) [inline src/main/resources/messages/user_messages.properties]: I think user message may be like "Internal system error" as well. A user should not even know what Cognito is.

File: src/main/java/gov/ca/cwds/idm/service/IdmServiceImpl.java (hunk 1 above)
Comment (dmytro-palczewski-osi) [reply src/main/java/gov/ca/cwds/idm/service/IdmServiceImpl.java]: We already have
  public static final String DATE_FORMAT = "yyyy-MM-dd";
  public static final String TIME_FORMAT = "HH:mm:ss";
  public static final String DATE_TIME_FORMAT = DATE_FORMAT + " " + TIME_FORMAT;
in Utils 
and it is not the same as date format used in IdmResource

File: src/main/resources/messages/user_messages.properties (hunk 2 above)
Comment (dmytro-palczewski-osi) [reply src/main/resources/messages/user_messages.properties]: Changed to "CAP internal error"

File: src/main/java/gov/ca/cwds/idm/service/IdmServiceImpl.java (hunk 1 above)
Comment (dmytro-palczewski-osi) [reply src/main/java/gov/ca/cwds/idm/service/IdmServiceImpl.java]: renamed to URL_DATETIME_FORMAT_PATTERN and URL_DATETIME_FORMATTER and moved to Utils class

**Pull request merged**